omod/src/main/java/org/openmrs/module/spa/servlet/SpaServlet.java

```diff
@@ -178,4 +234,24 @@ private void addCacheControlHeader(HttpServletRequest request, HttpServletRespon
         }
     }
 
+    private File getJsonConfigFile() {
+        File folder = SpaModuleUtils.getSpaStaticFilesDir();
+        if (!folder.exists()) {
+            folder.mkdirs();
+        }
+        if (!folder.isDirectory()) {
+            throw new ModuleException("SPA frontend repository is not a directory at: " + folder.getAbsolutePath());
+        }
+
+        AdministrationService as = Context.getAdministrationService();
+        String configFolderName = as.getGlobalProperty(CONFIG_DIRECTORY, DEFAULT_CONFIG_DIRECTORY);
+        String jsonConfigFileName = as.getGlobalProperty(JSON_CONFIG_FILE_NAME, DEFAULT_JSON_CONFIG_FILE_NAME);
+
+        File configDir = new File(folder +"/"+ configFolderName);
+        if (!configDir.exists()) {
+            configDir.mkdirs(); // create directory and all parent directories if necessary
+        }
```

Comment:
This seems like something we should only do when actually writing the file... Currently, we're also creating directories on a read?